kernel/rtlil.cc

```diff
@@ -445,6 +445,27 @@ vector<string> RTLIL::AttrObject::get_hdlname_attribute() const
 	return split_tokens(get_string_attribute(ID::hdlname), " ");
 }
 
+void RTLIL::AttrObject::set_intvec_attribute(RTLIL::IdString id, const vector<int> &data)
+{
+	std::stringstream attrval;
+	for (auto &i : data) {
+		if (attrval.tellp() > 0)
+			attrval << " ";
+		attrval << i;
+	}
+	attributes[id] = RTLIL::Const(attrval.str());
+}
+
+vector<int> RTLIL::AttrObject::get_intvec_attribute(RTLIL::IdString id) const
+{
+	vector<int> data;
+	auto it = attributes.find(id);
+	if (it != attributes.end())
+		for (const auto &s : split_tokens(attributes.at(id).decode_string()))
+			data.push_back(atoi(s.c_str()));
```

Comment:
Can we please not make Yosys have even more poorly defined type conversions? This one actually executes UB if the input is out of range, and also has different behavior on 32-bit and 64-bit platforms. Take a look at the RTLIL lexer to see how this can be done safely.

Reply:
I more or less copied straight from that; hopefully that's good enough.